src/main/java/org/jfrog/hudson/pipeline/common/executors/PublishBuildInfoExecutor.java

```diff
@@ -14,17 +15,20 @@
     private TaskListener listener;
     private BuildInfo buildInfo;
     private Run build;
+    private FilePath ws;
 
-    public PublishBuildInfoExecutor(Run build, TaskListener listener, BuildInfo buildInfo, ArtifactoryServer pipelineServer) {
+    public PublishBuildInfoExecutor(Run build, TaskListener listener, BuildInfo buildInfo, ArtifactoryServer pipelineServer, FilePath ws) {
         this.pipelineServer = pipelineServer;
         this.buildInfo = buildInfo;
         this.listener = listener;
         this.build = build;
+        this.ws = ws;
     }
 
     @Override
     public void execute() throws Exception {
         BuildInfoAccessor buildInfoAccessor = new BuildInfoAccessor(buildInfo);
+        buildInfoAccessor.appendVcs(Utils.extractVcs(ws));
```

Comment:
You already have access to the `Run` where you can access the build's ENV

Somewhat 😢 that publishBuildInfo requires an agent to publish this step.

```java
EnvVars env = build.getEnvironment(listener)
env.get("GIT_URL")
```

You should be able to loop over `GIT_URL_1`, `GIT_URL_2`

See

https://github.com/jenkinsci/git-plugin/blob/2e6d1bf0d10e04632f8153a3d2d9fc4d7d7c8bf2/src/main/java/hudson/plugins/git/GitSCM.java#L1368-L1376